lambdas/indexer/.chalice/config.json.template.py

```diff
@@ -25,7 +25,6 @@
     "manage_iam_role": False,
     "iam_role_arn": "${aws_iam_role.%s.arn}" % app_name,
     "environment_variables": config.lambda_env,
-    "minimum_compression_size": config.minimum_compression_size,
```

Comment:
Please provide evidence that there isn't a residual, and potentially conflicting entry in the resulting TF config generated by Chalice.

Reply:
## Property present in TF config built from `develop` branch
```
azul2 on  develop  daniel.local 🐍 v3.11.9 (.venv)
at 08/01 10:24:32 ❯ git co develop
Already on 'develop'
Your branch is up to date with 'github/develop'.

azul2 on  develop  daniel.local 🐍 v3.11.9 (.venv)
at 08/01 10:24:37 ❯ git log -n 1
commit 3bc6669d4ec962ac14ce4e8714871539624d7ae4 (HEAD -> develop, github/develop, github/HEAD)
Merge: 7ea9749d7 36ee7e2c9
Author: Daniel Sotirhos <dsotirho@ucsc.edu>
Date:   Wed Jul 31 10:55:14 2024 -0700

    Content Security Policy (CSP) Not Implemented (DataBiosphere/azul-private#6, #6384)

azul2 on  develop  daniel.local 🐍 v3.11.9 (.venv)
at 08/01 10:29:25 ❯ cd lambdas/

azul2/lambdas on  develop  daniel.local 🐍 v3.11.9 (.venv)
at 08/01 10:29:42 ❯ make
/Library/Developer/CommandLineTools/usr/bin/make -C layer layer
…
Creating .chalice/config.json
python vendor/resources/environ.json.template.py vendor/resources/environ.json
Creating vendor/resources/environ.json
PYTHONHASHSEED=0 python -m compileall \
        -f -q --invalidation-mode checked-hash \
        vendor vendor/humancellatlas vendor/azul
python -m azul.changelog vendor
chalice package --stage daniel --pkg-format terraform .chalice/terraform
/Users/daniel/repo/azul2/.venv/lib/python3.11/site-packages/_distutils_hack/__init__.py:32: UserWarning: Setuptools is replacing distutils. Support for replacing an already imported distutils is deprecated. In the future, this condition will fail. Register concerns at https://github.com/pypa/setuptools/issues/new?template=distutils-deprecation.yml
  warnings.warn(
Creating deployment package.
Reusing existing deployment package.
[INFO]  2024-08-01T10:29:56.015Z    00010ca1-b0ba-466f-8c58-dabbad000000    botocore.credentials    Found credentials in environment variables.

azul2/lambdas on  develop  daniel.local 🐍 v3.11.9 (.venv) took 10s
at 08/01 10:29:56 ❯ grep compress */.chalice/terraform/chalice.tf.json
indexer/.chalice/terraform/chalice.tf.json:        "minimum_compression_size": 0
service/.chalice/terraform/chalice.tf.json:        "minimum_compression_size": 0
```

## Property absent in TF config built from PR branch
```
azul2 on  develop  daniel.local 🐍 v3.11.9 (.venv)
at 08/01 10:30:27 ❯ git co issues/dsotirho-ucsc/6360-compressed-responses
Switched to branch 'issues/dsotirho-ucsc/6360-compressed-responses'
Your branch is up to date with 'github/issues/dsotirho-ucsc/6360-compressed-responses'.

azul2 on  issues/dsotirho-ucsc/6360-compressed-responses  daniel.local 🐍 v3.11.9 (.venv)
at 08/01 10:30:37 ❯ git log -n 1
commit c5ea7566f987d4a19ee634104825cdd9578aa0c7 (HEAD -> issues/dsotirho-ucsc/6360-compressed-responses, github/issues/dsotirho-ucsc/6360-compressed-responses)
Author: Daniel Sotirhos <dsotirho@ucsc.edu>
Date:   Thu Aug 1 10:18:37 2024 -0700

    fixup! Fix: Service responses are not compressed (#6360)

azul2 on  issues/dsotirho-ucsc/6360-compressed-responses  daniel.local 🐍 v3.11.9 (.venv)
at 08/01 10:30:43 ❯ cd lambdas/

azul2/lambdas on  issues/dsotirho-ucsc/6360-compressed-responses  daniel.local 🐍 v3.11.9 (.venv)
at 08/01 10:30:51 ❯ make
/Library/Developer/CommandLineTools/usr/bin/make -C layer layer
…
Creating .chalice/config.json
python vendor/resources/environ.json.template.py vendor/resources/environ.json
Creating vendor/resources/environ.json
PYTHONHASHSEED=0 python -m compileall \
        -f -q --invalidation-mode checked-hash \
        vendor vendor/humancellatlas vendor/azul
python -m azul.changelog vendor
chalice package --stage daniel --pkg-format terraform .chalice/terraform
/Users/daniel/repo/azul2/.venv/lib/python3.11/site-packages/_distutils_hack/__init__.py:32: UserWarning: Setuptools is replacing distutils. Support for replacing an already imported distutils is deprecated. In the future, this condition will fail. Register concerns at https://github.com/pypa/setuptools/issues/new?template=distutils-deprecation.yml
  warnings.warn(
Creating deployment package.
Reusing existing deployment package.
[INFO]  2024-08-01T10:31:02.672Z    00010ca1-b0ba-466f-8c58-dabbad000000    botocore.credentials    Found credentials in environment variables.

azul2/lambdas on  issues/dsotirho-ucsc/6360-compressed-responses  daniel.local 🐍 v3.11.9 (.venv) took 9s
at 08/01 10:31:02 ❯ grep compress */.chalice/terraform/chalice.tf.json

azul2/lambdas on  issues/dsotirho-ucsc/6360-compressed-responses  daniel.local 🐍 v3.11.9 (.venv)
at 08/01 10:31:25 ❌1 ❯
```

Reply:
I'm having a hard time parsing this, probably due to the elaborate prompt setup you have going there. We usually use `$` to indicate a shell prompt in terminal transcripts. Could you simplify your evidence according to that convention. The `*/.chalice/terraform/chalice.tf.json` files are intermediates, what counts ultimately are the `tf.json` files in `terraform/`. Please use those. Lastly, please use `jq` to extract the resource that *would* contain the conflicting property.

Reply:
Scratch that. In fact, since we're post-processing the Chalice-generated TF config, please just add an assertion that the corresponding TF resource property is absent. Then we won't need any evidence.